src/components/clusters/ClusterDetailsMultiRegion/components/IdentityProvidersPage/components/HtpasswdDetails/UploadHTPasswdFileModal.test.tsx

```
@@ -0,0 +1,193 @@
+};
+
+const validFileContent = 'user1:$2y$05$hash1\nuser2:$2y$05$hash2';
+const invalidFileContent = 'invalidline\nuser1:pass1';
+
+const uploadFile = async (content: string, filename = 'users.htpasswd') => {
+  const file = new File([content], filename, { type: 'text/plain' });
+  const fileInput = document.querySelector('input[type="file"]') as HTMLInputElement;
+  await waitFor(() =>
+    fireEvent.change(fileInput, {
+      target: { files: [file] },
+    }),
```

ESLint

[error] Avoid using side effects within `waitFor` callback

(testing-library/no-wait-for-side-effects)